### ISSUE TITLE
Use the `padding` arg when possible for torch backend

### ISSUE THREAD

**Issue by haifeng-jin** [keras-team/keras#18657]
To support `padding="same"`, we implemented a custom padding by calling `torch.nn.functional.pad` for ops including `conv`, `max_pool`, and `average_pool`.

However, it takes more memory and slows down the model.

Torch does support some cases of `padding="same"`.
We should only use this manuall padding method when the native ops in torch doesn't support it.
